### ISSUE TITLE
Preventing external requests

### ISSUE THREAD

**Issue by K73SK** [golang/go#39253]
# What version of Go are you using (`go version`)?
1.14

### Does this issue reproduce with the latest release?
N/A

### What operating system and processor architecture are you using (`go env`)?

Windows & Mac 64-bit each

### What did you do?

tried to set all the environment variables I could think of

### What did you expect to see?

To not reach externally

### What did you see instead?

To reach only internally

------------

I'm working at a company where we are trying to bring GoLang in. One of the requirements by security is that it cannot reach externally to sites like github.com, gitlab.com, etc. when users request for new modules. Another situation that I've been told to make sure we avoid is users doing just `go get` and updating to the latest version of `go` automatically that way. So commands like this:

```
go get github.com/group/project
```

Should not pull in the module from github.com. The request is that it pulls in the resource from an internal repository instance that we have setup. 

I've tried to setup the variables GOPROXY, GONOPROXY, and GOPRIVATE, but none are working unless I specifically put the modules in the go.mod file.

**In summary**, how can I setup `go get` to only ever point to the internal repository 100% of the time, and return a 404 when a specific module supplied does not exist? Note: we cannot rely on the `.gitconfig` file since there's never a guarantee that only github.com and gitlab.com are where the resources are pulled from.

**Comment by andybons** [comment]
Hi there,

Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and [proposals](https://golang.org/s/proposal-process) only.

For asking questions, see:

* [Stack Overflow](https://stackoverflow.com/questions/tagged/go) with questions tagged "go"

* [The Go Forum](https://forum.golangbridge.org/), a web-based forum

* [Gophers Slack](https://gophers.slack.com), use the [invite app](https://invite.slack.golangbridge.org/) for access. The `#general` channel is a good starting point.

* [Go Community on Hashnode](https://hashnode.com/n/go) with questions and posts tagged with "go"

* [The golang-nuts mailing list](https://groups.google.com/d/forum/golang-nuts)

* **IRC** channel #go-nuts on Freenode

Please ask the question on one of the above forums.

(Quoted from https://golang.org/wiki/Questions)
